de.fu_berlin.inf.dpp.whiteboard/src/de/fu_berlin/inf/dpp/whiteboard/ui/browser/BrowserActionExecuter.java

```diff
@@ -0,0 +1,101 @@
+package de.fu_berlin.inf.dpp.whiteboard.ui.browser;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.log4j.Logger;
+
+import de.fu_berlin.inf.dpp.whiteboard.sxe.SXEController;
+import de.fu_berlin.inf.dpp.whiteboard.sxe.exceptions.MissingRecordException;
+import de.fu_berlin.inf.dpp.whiteboard.sxe.records.AttributeRecord;
+import de.fu_berlin.inf.dpp.whiteboard.sxe.records.ElementRecord;
+import de.fu_berlin.inf.dpp.whiteboard.sxe.records.IRecord;
+import de.fu_berlin.inf.dpp.whiteboard.ui.browser.BrowserAction.ActionTypes;
+
+public class BrowserActionExecuter {
+
+    private static final Logger LOG = Logger
+        .getLogger(BrowserActionExecuter.class);
+
+    private final SXEController sxe;
+
+    public BrowserActionExecuter(SXEController sxe) {
+        this.sxe = sxe;
+    }
+
+    public void execute(BrowserAction action) {
+        sxe.executeAndCommit(getChangeRecords(action));
+    }
+
+    private List<IRecord> getChangeRecords(BrowserAction action) {
+        String type = action.getType();
+        if (type.equals(ActionTypes.NEW)) {
+            return create(action);
+        } else if (type.equals(ActionTypes.SET)) {
+            return set(action);
+        } else if (type.equals(ActionTypes.REMOVE)) {
+            return remove(action);
+        } else {
+            LOG.error("undefined action type received: " + type);
+            return new LinkedList<IRecord>();
+        }
+    }
+
+    private List<IRecord> create(BrowserAction action) {
+        List<IRecord> records = new LinkedList<IRecord>();
```

Comment:
ArrayList should be sufficient, this applies to all appearances in this patch set.